thunder/core/proxies.py

```diff
@@ -1239,6 +1242,8 @@ def _infer_tensor_properties(
     _dtype = dtypes.numbertype_to_dtype(_dtype) if dtypes.is_numbertype(_dtype) else _dtype
     _requires_grad = requires_grad if requires_grad is not None else _requires_grad
     _requires_grad = False if not dtypes.is_inexact_dtype(_dtype) else _requires_grad
+    _grad = grad if grad is not None else _grad
+    _grad = None if not _requires_grad else _grad
```

Comment:
So `_grad` still can be `None`, which I think possible for multiple cases?

Reply:
I did not find `TensorProxy.grad` being `None` to be problematic. For the purpose of tracing an optimizer's `_init_group`, Thunder needs to understand the idiom `if p.grad is not None:` ([`torch/optim/adam.py`](https://github.com/pytorch/pytorch/blob/30b98940b8c4f2903110fe497705ff62df5e968b/torch/optim/adam.py#L126)).